examples/scripts/reward_trainer.py

```diff
@@ -77,11 +79,14 @@ class ScriptArguments:
     quantization_config=quantization_config,
     device_map=device_map,
     trust_remote_code=script_args.trust_remote_code,
+    num_labels=1,
 )
 
 # Step 2: Load the dataset and pre-process it
 tokenizer = AutoTokenizer.from_pretrained(script_args.model_name)
-dataset = load_dataset(script_args.dataset_name, split="train")
+train_dataset = load_dataset(script_args.dataset_name, split="train")
+eval_dataset = load_dataset(script_args.dataset_name, split="test")
```

Comment:
I think this should be optional - not every dataset has a `test` split. Maybe we can pass a `eval-split` to the cli arguments. by default it is none (no eval set used) and alternatively pass a split (e.g. `validation` or `test`). 

what do you think?

Reply:
Good point. Done.